File: web/test_chat.py
import urllib.request
import json
import sys

url = 'http://localhost:3000/api/ai-chat'
data = json.dumps({
    'messages': [{'role': 'user', 'content': 'حلل ABUK هات أخباره لو كسر الدعم أعمل إيه؟'}]
}).encode('utf-8')

req = urllib.request.Request(url, data=data, headers={'Content-Type': 'application/json'}, method='POST')
try:
    with urllib.request.urlopen(req) as response:
        print(f"STATUS: {response.getcode()}")
        while True:
            chunk = response.read(1024)
            if not chunk:
                break
            print(f"CHUNK: {chunk.decode('utf-8', errors='replace')}")
except Exception as e:
    print(f"Error: {e}")

File: web/test_chat.js
const http = require('http');

const data = JSON.stringify({
  message: 'حلل ABUK هات أخباره لو كسر الدعم أعمل إيه؟',
  history: []
});

const options = {
  hostname: '127.0.0.1',
  port: 3000,
  path: '/api/ai-chat',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': Buffer.byteLength(data)
  }
};

const req = http.request(options, (res) => {
  console.log(`STATUS: ${res.statusCode}`);
  res.setEncoding('utf8');
  res.on('data', (chunk) => {
    console.log(`CHUNK: ${chunk}`);
  });
});

req.on('error', (e) => {
  console.error(`problem with request: ${e.message}`);
});

req.write(data);
req.end();
